services/web/server/src/simcore_service_webserver/director_v2_core.py

```diff
@@ -441,38 +420,46 @@ async def get_service_state(app: web.Application, node_uuid: str) -> DataType:
 
 
 @log_decorator(logger=log)
-async def retrieve(
-    app: web.Application, node_uuid: str, port_keys: List[str]
-) -> DataBody:
-    # when triggering retrieve endpoint
-    # this will allow to sava bigger datasets from the services
-    # TODO: PC -> ANE: all settings MUST be in app[APP_SETTINGS_KEY]
-    timeout = ServicesCommonSettings().storage_service_upload_download_timeout
-
+async def retrieve_dynamic_service_inputs(
+    app: web.Application, service_uuid: str, port_keys: List[str]
+) -> DataType:
+    """Pulls data from connections to the dynamic service inputs"""
```

Comment:
@GitHK : question, how so not used? how do you trigger the retrieve inputs when the outputs of the preceeding services change? how does the dynamic-sidecar knows it? is it listening to the postgres DB?

Reply:
Made a bit of a mess. This call only pulls inputs in the case of the dynamic-sidecar.And forget the above comment.

I wanted to mention that when the dynamic-sidecar starts it will also pull the inputs automatically before starting the service's containers.
In the case of the legacy services this call would happen after the services started.

With the dynamic-sidecar, to guarantee consistency, the data is pulled before starting the service. So the user already has it in his environment when the service starts.